fig.go

```diff
@@ -28,6 +28,42 @@ const (
 	DefaultTimeLayout = time.RFC3339
 )
 
+// StringUnmarshaler is an interface for custom unmarshaling of strings
+//
+// If a field with a local type asignment satisfies this interface, it allows the user
+// to implment their own custom type unmarshaling method.
+//
+// Example:
```

Comment:
Love the comment! 

It would be great to also add a working example under `/examples`.

Reply:
Will do!